src/resource/file.ts

```diff
@@ -305,7 +319,15 @@ async function writeFile(
   if (config.slug !== undefined) {
     headers["Slug"] = config.slug;
   }
-  headers["Content-Type"] = file.type;
+  const fileType =
+    typeof file === "object" &&
+    file !== null &&
+    typeof (file as Blob).type === "string" &&
+    (file as Blob).type.length > 0
+      ? (file as Blob).type
+      : undefined;
+  headers["Content-Type"] =
+    options.contentType ?? fileType ?? "application/octet-stream";
```

Comment:
This (rather intricate) piece of logic is duplicated. Should it be isolated in its own function ?

Reply:
Hmm yeah good point, it started out rather innocently, but kinda grew into this monster without me noticing. I'll refactor it.

Reply:
ecb193fc7cdba8d0935e1a08adc58fc93f6a0f62